hermes/crates/cardano-chain-follower/examples/follow_chains.rs

```diff
@@ -254,24 +258,16 @@ async fn follow_for(network: Network, matches: ArgMatches) {
                     dump_raw_aux_data = true;
                 }
 
+                // If flag `bad_cip36` is set, log the CIP36 validation.
                 if bad_cip36 {
-                    #[allow(irrefutable_let_patterns)] // Won't always be irrefutable.
-                    if let Metadata::DecodedMetadataValues::Cip36(cip36) = &decoded_metadata.value {
-                        if !cip36.signed {
-                            dump_raw_aux_data = true;
-                        }
-                        if !decoded_metadata.report.is_empty() {
-                            info!(
-                                chain = network.to_string(),
-                                "Cip36 {tx_idx}:{:?} - {raw_size}", decoded_metadata
-                            );
-                            dump_raw_aux_data = true;
-                        }
-                    }
+                    dump_raw_aux_data = cip36(&decoded_metadata, network, tx_idx, raw_size);
```

Comment:
If bad error occurs in CIP36 data, will dump the raw auxiliary data for debugging purpose. 
@stevenj  include it for CIP36, but I don't think it necessary for CIP509 since it will dump a huge bytes data

Reply:
Refactor the code and add `log-raw-aux`